### PR TITLE
Implement rejectDate option

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
Thanks @MaxGraey - can you also add the test cases that relate to this feature? I think the following should now work:

https://github.com/ColinEberhardt/assemblyscript-temporal/blob/master/assembly/__tests__/plaindate.spec.ts#L1116-L1120

**Comment by MaxGraey** [comment]
@ColinEberhardt done!

File: assembly/__tests__/plaindate.spec.ts [hunk 1]
```diff
@@ -1112,13 +1114,13 @@ describe("Date.equal works", () => {
...
-//   it('constructing from numbers', () => {
-//     throws(() => new PlainDate(-271821, 4, 18), RangeError);
-//     throws(() => new PlainDate(275760, 9, 14), RangeError);
-//     expect(new PlainDate(-271821, 4, 19).toString()).toBe('-271821-04-19');
-//     expect(new PlainDate(275760, 9, 13).toString()).toBe('+275760-09-13');
-//   });
+describe('Min/max range', () => {
+  it('constructing from numbers', () => {
+    // throws("RangeError", () => { new PlainDate(-271821, 4, 18) });
+    // throws("RangeError", () => { new PlainDate(275760, 9, 14) });
```
Comment (MaxGraey) [inline assembly/__tests__/plaindate.spec.ts:1120]: this still not work. Need to implement epochFromParts properly

Comment (ColinEberhardt) [reply assembly/__tests__/plaindate.spec.ts:1120]: I presume you could use this? https://github.com/ColinEberhardt/assemblyscript-temporal/blob/master/assembly/date.ts#L155

Comment (MaxGraey) [reply assembly/__tests__/plaindate.spec.ts:1120]: Unfortunately no. `epochFromParts` it seems required `u128` arithmetic (bignum sum)

Comment (MaxGraey) [reply assembly/__tests__/plaindate.spec.ts:1120]: we can use as-bignum but probably we could avoid this. Need to think over

**Pull request merged**